src/battle.ts

```diff
@@ -2984,7 +2996,7 @@ class Battle {
 			let pokemon = side.pokemon[i];
 			if (pokemon.fainted) continue;
 			// already active, can't be switching in
-			if (side.active.includes(pokemon)) continue;
+			if (side.active.includes(pokemon) && side.ally) continue;
```

Comment:
What is this intended to do? I don't think it does what it's intended to.

Reply:
so currently, it you cant click on your second pokemon which is not active, because it thinks its active, because your teammate has an active pokemon. adding with willl make sure you can switch to your second pokemon, since his way the client knows its an ally and wont make the second pokemon unclickable.
should probably be ```!side.ally```

Reply:
That's not what this code does...

Reply:
cause what i believe, side.active includes your allys pokemon rn notsure

Reply:
i study this deeper